token/parse_test.go

```diff
@@ -290,20 +291,22 @@ func TestPayload_Type(t *testing.T) {
 		Google *GCPGooglePayload
 		Amazon *AWSAmazonPayload
 		Azure  *AzurePayload
+		Claims *jose.Claims
 	}
 	tests := []struct {
 		name   string
 		fields fields
 		want   Type
 	}{
-		{"JWK", fields{"a-sha", []string{"foo.bar.zar"}, "", nil, nil, nil}, JWK},
-		{"JWK no sans", fields{"a-sha", nil, "", nil, nil, nil}, JWK},
-		{"JWK no sha", fields{"", []string{"foo.bar.zar"}, "", nil, nil, nil}, JWK},
-		{"OIDC", fields{"", nil, "mariano@smallstep.com", nil, nil, nil}, OIDC},
-		{"GCP", fields{"", nil, "", &GCPGooglePayload{}, nil, nil}, GCP},
-		{"AWS", fields{"", nil, "", nil, &AWSAmazonPayload{}, nil}, AWS},
-		{"Azure", fields{"", nil, "", nil, nil, &AzurePayload{}}, Azure},
-		{"Unknown", fields{"", nil, "", nil, nil, nil}, Unknown},
+		{"JWK", fields{"a-sha", []string{"foo.bar.zar"}, "", nil, nil, nil, nil}, JWK},
+		{"JWK no sans", fields{"a-sha", nil, "", nil, nil, nil, nil}, JWK},
+		{"JWK no sha", fields{"", []string{"foo.bar.zar"}, "", nil, nil, nil, nil}, JWK},
+		{"OIDC", fields{"", nil, "mariano@smallstep.com", nil, nil, nil, nil}, OIDC},
+		{"GCP", fields{"", nil, "", &GCPGooglePayload{}, nil, nil, nil}, GCP},
+		{"AWS", fields{"", nil, "", nil, &AWSAmazonPayload{}, nil, nil}, AWS},
+		{"Azure", fields{"", nil, "", nil, nil, &AzurePayload{}, nil}, Azure},
+		{"Unknown", fields{"", nil, "", nil, nil, nil, nil}, Unknown},
+		{"OIDC Kubernetes", fields{"", nil, "", nil, nil, nil, &jose.Claims{Audience: jwt.Audience{"step-ca"}, Issuer: "https://kubernetes.default.svc.cluster.local", Subject: "system:serviceaccount:default:default", Expiry: jwt.NewNumericDate(time.Now()), IssuedAt: jwt.NewNumericDate(time.Now())}}, OIDC},
```

Comment:
You can use or wrapper and use `jose.NewNumericDate()` it will be easier to move to go-jose/v3

Reply:
948d88c4